lib/core/ThroughputLimiter.ts

```diff
@@ -0,0 +1,48 @@
+import IVersionManager from './interfaces/IVersionManager';
+import TransactionModel from '../common/models/TransactionModel';
+
+/**
+ * Keeps track of current block and throughput limits based on the state
+ */
+export default class ThroughputLimiter {
+
+  private currentBlockHeight: number | undefined;
+  private transactionsInCurrentBlock: TransactionModel[];
+
+  constructor (
+    private versionManager: IVersionManager
+  ) {
+    this.transactionsInCurrentBlock = [];
+  }
+
+  /**
+   * given a an array of transactions, return an array of qualified transactions based on current state
+   * @param transactions array of transactions to filter for
+   */
+  public async getQualifiedTransactions (transactions: TransactionModel[]) {
+    let qualifiedTransactions: TransactionModel[] = [];
+    for (const transaction of transactions) {
+      if (transaction.transactionTime === this.currentBlockHeight) {
+        this.transactionsInCurrentBlock.push(transaction);
+      } else {
+        if (this.currentBlockHeight !== undefined) {
```

Comment:
Design: it just occurred to me that it is quite undesirable for transactions to be held indefinitely in the priority queue, until if/when another transaction is written in a later block... this seems to be a real issue at least during the foreseeable future. The only way I can think of to remedy this is to require a logical interface change to blockchain `read()` to always return complete block data. Unfortunately I think it is the right thing to do, what do you guys think?

Reply:
Or maybe we can put them in something like MongoDB until they are needed? I assume memory is your concern right?

Reply:
Will address in another issue and pr